### PR TITLE
Linguist-backend: Add token to catalogClient

### PR DESCRIPTION
**Pull request by anicke**
## Hey, I just made a Pull Request!

Added the token option to the catalogClient calls. This is necessary for Backstage deployments
that uses the token authentication.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/linguist-backend/api-report.md [hunk 1]
```diff
@@ -96,6 +98,8 @@ export interface RouterOptions {
   reader: UrlReader;
   // (undocumented)
   scheduler?: PluginTaskScheduler;
+  // (undocumented)
+  tokenManager: TokenManager;
```
Comment (freben) [inline plugins/linguist-backend/api-report.md:102]: This is technically a breaking change, so you could either make it optional (and handle the complexity of that in code), or make the changeset `minor` level. The changeset should also probably say explicitly that it added a token manager as the dependency to the plugin, so as to be informative to the reader who wonders what the effect is on them as they are upgrading.

File: plugins/linguist-backend/src/api/LinguistBackendApi.ts [hunk 2]
```diff
@@ -46,6 +50,8 @@ export class LinguistBackendApi {
   private readonly store: LinguistBackendStore;
   private readonly urlReader: UrlReader;
   private readonly discovery: PluginEndpointDiscovery;
+  private readonly tokenManager: TokenManager;
+
```
Comment (freben) [inline plugins/linguist-backend/src/api/LinguistBackendApi.ts:54]: nit: extra newline ;) wouldn't have commented if there wasn't other things, but...

File: plugins/linguist-backend/src/api/LinguistBackendApi.ts [hunk 3]
```diff
@@ -98,7 +106,8 @@ export class LinguistBackendApi {
       fields: ['kind', 'metadata'],
     };
 
-    const response = await this.catalogClient.getEntities(request);
+    const { token } = await this.tokenManager.getToken();
+    const response = await this.catalogClient.getEntities(request, { token });
```
Comment (freben) [inline plugins/linguist-backend/src/api/LinguistBackendApi.ts:110]: if you do make the token manager an optional thing to supply (which actually can be helpful in tests too i guess), you could make it like

```ts
const tr = await this.tokenManager?.getToken();
const response = await this.catalogClient.getEntities(request, { token: tr?.token });
```

File: plugins/linguist-backend/src/api/LinguistBackendApi.ts [hunk 4]
```diff
@@ -117,8 +126,11 @@ export class LinguistBackendApi {
       0,
       this.batchSize ?? 20,
     );
+    const { token } = await this.tokenManager.getToken();
```
Comment (freben) [inline plugins/linguist-backend/src/api/LinguistBackendApi.ts:129]: Theoretically, you're meant to call getToken for every single request, ie inside the forEach. It's meant to be caching and fast, so there's less risk for sudden expiry in the midst of ongoing work if it takes some time to finish.

**Review by freben — COMMENTED** [review]
Awesome! Just some small notes

File: plugins/linguist-backend/api-report.md (hunk 1 above)
Comment (anicke) [reply plugins/linguist-backend/api-report.md:102]: I've marked the change as "breaking". In the install instructions all of the "PluginEnvironment" is spread as the RouterOptions so there will probably be no practical impact when upgrading.

File: plugins/linguist-backend/src/api/LinguistBackendApi.ts (hunk 4 above)
Comment (anicke) [reply plugins/linguist-backend/src/api/LinguistBackendApi.ts]: Done!

**Review by freben — COMMENTED** [review]
Nice!

**Review by freben — APPROVED** [review]

File: plugins/linguist-backend/api-report.md (hunk 1 above)
Comment (freben) [reply plugins/linguist-backend/api-report.md:102]: Yeah it'll depend on people's individual setups but in practice most probably have followed those.

**Pull request merged**